clang/utils/TableGen/ClangAttrEmitter.cpp

```diff
@@ -4091,13 +4100,15 @@ EmitClangAttrTemplateInstantiateHelper(ArrayRef<const Record *> Attrs,
     for (auto const &ai : Args)
       ai->writeTemplateInstantiation(OS);
 
-    OS << "      return new (C) " << R.getName() << "Attr(C, *A";
+    OS << "      auto* Result = new (C) " << R.getName() << "Attr(C, *A";
     for (auto const &ai : Args) {
       OS << ", ";
       ai->writeTemplateInstantiationArgs(OS);
     }
-    OS << ");\n"
-       << "    }\n";
+    OS << ");\n";
+    OS << R.getValueAsString("PostInstantiationStmts");
+    OS << "return Result;\n";
```

Comment:
Can you share what is going on here a bit better?  It isn't clear what the intent is here without examples.